### PR TITLE
Menus

### PR DESCRIPTION
**Pull request by kasperg**
This PR adds support for letting administrators work with the site menus and allow these menus to be exported through the REST API.

### CODE REVIEW

**Comment by kasperg** [comment]
@kallehauge This is now ready for review.

File: web/modules/custom/menu_rest_resource/src/Plugin/Rest/Resource/MenuResource.php [hunk 1]
```diff
@@ -0,0 +1,188 @@
+<?php
```
Comment (kallehauge) [inline web/modules/custom/menu_rest_resource/src/Plugin/Rest/Resource/MenuResource.php]: This file is identical to `web/modules/custom/menu_rest_resource/src/Plugin/rest/resource/MenuResource.php`

**Comment by kallehauge** [comment]
Besides a duplicate file, then this looks great and is ready to get merged :+1:

**Comment by kasperg** [comment]
Argh - now I see why Scrutinizer complained about duplicate files :+1:

That should be fixed now.

**Comment by kallehauge** [comment]
There is a merge conflict. Feel free to merge the branch when this have been resolved :)

**Pull request merged**